wandb/sdk/data_types/table.py

```diff
@@ -557,6 +557,91 @@ def bind_to_run(self, *args, **kwargs):
+
+        Returns:
+            List of table rows from all increments
+        """
+        if "latest" not in source_artifact.aliases:
+            wandb.termwarn(
+                "It is recommended to use the latest version of the incremental table artifact for ordering guarantees.",
+                repeat=False,
+            )
+        data: List[Any] = []
+        increment_num = json_obj.get("increment_num", None)
+        if increment_num is None:
+            return data
+
+        increment_entries = {
+            entry_key: source_artifact.manifest.entries[entry_key]
+            for entry_key in source_artifact.manifest.entries
+            if not entry_key.startswith("media") and entry_key.endswith("table.json")
+        }
+
+        # Sort by increment number first, then by timestamp if present
+        # Format of  is: "{incr_num}-{timestamp_ms}.{key}.table.json"
+        def get_sort_key(key: str) -> tuple:
+            parts = key.split(".")
+            increment_parts = parts[0].split("-")
+            increment_num = int(increment_parts[0])
+            # If there's a timestamp part, use it for secondary sorting
+            timestamp = int(increment_parts[1]) if len(increment_parts) > 1 else 0
+            return (increment_num, timestamp)
+
+        sorted_increment_keys = sorted(increment_entries.keys(), key=get_sort_key)
+
+        for entry_key in sorted_increment_keys:
+            with open(increment_entries[entry_key].download()) as f:
+                table_data = json.load(f)
+                data.extend(table_data["data"])
+
+        return data
+
+    @classmethod
+    def _process_table_row(
+        cls,
+        row: List[Any],
+        timestamp_column_indices: set,
+        np_deserialized_columns: Dict[int, Any],
+        source_artifact,
+        row_idx: int,
+    ) -> List[Any]:
+        """Process a single row of table data.
+
+        Args:
+            row: The row data to process
+            timestamp_column_indices: Set of column indices containing timestamps
+            np_deserialized_columns: Dictionary mapping column indices to numpy arrays
+            source_artifact: The source artifact containing the table data
+            row_idx: The index of the current row
+
+        Returns:
+            Processed row data
+        """
+        row_data = []
+        for c_ndx, item in enumerate(row):
+            cell = item
+            if c_ndx in timestamp_column_indices and isinstance(item, (int, float)):
+                cell = datetime.datetime.fromtimestamp(
+                    item / 1000, tz=datetime.timezone.utc
+                )
+            elif c_ndx in np_deserialized_columns:
+                cell = np_deserialized_columns[c_ndx][row_idx]
+            elif isinstance(item, dict) and "_type" in item:
+                obj = WBValue.init_from_json(item, source_artifact)
+                if obj is not None:
+                    cell = obj
```

Comment:
Since we're on Python 3.8+, you can reduce nesting here:

```suggestion
            elif (
                isinstance(item, dict)
                and "_type" in item
                and (obj := WBValue.init_from_json(item, source_artifact))
            ):
                cell = obj
```